gtsfm/scene_optimizer.py

```diff
@@ -432,11 +443,62 @@ def save_full_frontend_metrics(
                 if report.reproj_error_gt_model is not None and report.v_corr_idxs_inlier_mask_gt is not None
                 else None,
                 "inlier_ratio_est_model": round_fn(report.inlier_ratio_est_model),
-                "num_inliers_est_model": report.num_inliers_est_model,
+                "num_inliers_est_model": int(report.num_inliers_est_model)
+                if report.num_inliers_est_model is not None
+                else None,
             }
         )
 
     io_utils.save_json_file(os.path.join(METRICS_PATH, filename), metrics_list)
 
     # Save duplicate copy of 'frontend_full.json' within React Folder.
     io_utils.save_json_file(os.path.join(REACT_METRICS_PATH, filename), metrics_list)
+
+    if matching_regime not in [ImageMatchingRegime.RETRIEVAL, ImageMatchingRegime.SEQUENTIAL_WITH_RETRIEVAL]:
+        return
+    if "VIEWGRAPH_2VIEW_REPORT" in filename:
+        # must come after two-view report file is written to disk in the Dask dependency graph.
+        _save_retrieval_two_view_metrics()
+
+
+def _save_retrieval_two_view_metrics() -> None:
+    """Compare 2-view similarity scores with their 2-view pose errors after viewgraph estimation."""
```

Comment:
I am not sure how the pose errors are expected to capture visual similarity in image, especially the translation error. 
- cameras may be far away looking at the same object. 
- repetitive structure can have good visual similarity but high pose error.

Reply:
also why are we using the pose error from view graph estimation instead of ground truth? Im not sure if we want to evaluate a module using products downstream.

Reply:
so we are plotting error w.r.t. GT, for only the pairs that remain after viewgraph estimation, to get a better sense of how our 2-view errors are linked to the temporal distance in the sequence. doesn't make sense for all datasets, but for temporally ordered ones, i think it's still useful.

Reply:
I think it would be more informative to plot the results of all the pairs provided by the Retriever.

Reply:
Hi guys, thanks for the feedback here.

So I was interested in seeing how retrieval affects the final performance / back-end. If pairs dont have enough support, they never come out of the front-end, so they would just be noise in our system analysis. So I think the other plot could potentially lead to some incorrect conclusions.

Reply:
I still think visualizing the raw Retriever output would be the most useful. Having both may be even better to see how the solution evolves following the viewgraph?

Reply:
> So I was interested in seeing how retrieval affects the final performance / back-end. 
 
we can compare the final metrics with and without the retriever to evaluate that. 

this plot is a kind of validation that we are doing the right thing.. retrieved edges -> edges with low error in the view graph. but I don't think it is a metric that we want to monitor long-term. As Travis mentioned, it ignores the edges that were retrieved but were rejected during view graph estimation. 
If this is useful for debugging, we can have a separate viz script that reads these jsons and creates these plots, but doing that in the pipeline seems a little hacky to me

Reply:
please move this to a util file? feel free to create one if it doesn't exist already